lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -89,6 +70,27 @@ def fetch_logs
 
     private
 
+    def interpret_pod_status_data(status_data, metadata)
+      @status = @phase = (metadata["deletionTimestamp"] ? "Terminating" : status_data["phase"])
+
+      if @phase == "Failed" && status_data['reason'].present?
+        @status += " (Reason: #{status_data['reason']})"
+      elsif @phase != "Terminating"
+        ready_condition = status_data.fetch("conditions", []).find { |condition| condition["type"] == "Ready" }
+        @ready = ready_condition.present? && (ready_condition["status"] == "True")
+        @status += " (Ready: #{@ready})"
+      end
+    end
+
+    def log_suspicious_states(container_statuses)
+      container_statuses.each do |status|
+        waiting_state = status["state"]["waiting"] if status["state"]
+        reason = waiting_state["reason"] if waiting_state
+        next unless SUSPICIOUS_CONTAINER_STATES.include?(reason)
+        @logger.warn("#{id} has container in state #{reason} (#{waiting_state['message']})")
```

Comment:
I tophatted this by removing the failure stubbing in the `test_deploy_result_logging_for_mixed_result_deploy` test case (and adding `CrashLoopBackOff` to the suspicious states list--it should have been there). Result: 

![screenshot](https://screenshot.click/2017-06-27--195951_cku4u-z38bm.png)

It still works, and it is still noisy. I can't wait to replace this!